src/views/htmlcontent/src/js/components/app.component.spec.ts

```diff
@@ -66,54 +236,470 @@ describe('AppComponent', function (): void {
 
-        beforeEach(() => {
+    describe('basic behavior', () => {
+
+        beforeEach(async(() => {
             fixture = TestBed.createComponent<AppComponent>(AppComponent);
             fixture.detectChanges();
             comp = fixture.componentInstance;
             ele = fixture.nativeElement;
+        }));
+
+        it('should not hide message pane on click when there is no data', () => {
+            let messages = <HTMLElement> ele.querySelector('#messages');
+            expect(messages).not.toBeNull();
+            expect(messages.className.indexOf('hidden')).toEqual(-1, 'messages not visible');
+            messages.click();
+            fixture.detectChanges();
+            expect(messages.className.indexOf('hidden')).toEqual(-1, 'messages not visible');
         });
 
-        beforeEach(() => {
-            comp.dataService.webSocket.dispatchEvent(new MessageEvent('message', {
-                data: JSON.stringify(MockBatch1)
-            }));
+        it('should hide message pane on click when there is data', () => {
+            let dataService = <MockDataService> fixture.componentRef.injector.get(DataService);
+            dataService.sendWSEvent(batch2);
+            dataService.sendWSEvent(completeEvent);
+            fixture.detectChanges();
+            let messages = <HTMLElement> ele.querySelector('#messages');
+            expect(messages).not.toBeNull();
+            expect(messages.className.indexOf('hidden')).toEqual(-1, 'messages not visible');
+            let messagePane = <HTMLElement> ele.querySelector('#messagepane');
+            messagePane.click();
+            fixture.detectChanges();
+            expect(messages.className.indexOf('hidden')).not.toEqual(-1);
         });
 
-        it('should have initilized the grids correctly', () => {
+        it('should hide the results pane on click when there is data', () => {
+            let dataService = <MockDataService> fixture.componentRef.injector.get(DataService);
+            dataService.sendWSEvent(batch2);
+            dataService.sendWSEvent(completeEvent);
             fixture.detectChanges();
-            let results = ele.querySelector('#results');
+            let results = <HTMLElement> ele.querySelector('#results');
             expect(results).not.toBeNull('results pane is not visible');
-            expect(results.getElementsByTagName('slick-grid').length).toEqual(1);
-            // let slickgrid = results.getElementsByTagName('slick-grid')[0];
-            // let viewport = slickgrid.firstElementChild.getElementsByClassName('slick-viewport')[0]
+            expect(results.className.indexOf('hidden')).toEqual(-1);
+            let resultspane = <HTMLElement> ele.querySelector('#resultspane');
+            resultspane.click();
+            fixture.detectChanges();
+            expect(results.className.indexOf('hidden')).not.toEqual(-1);
+        });
+
+        it('should render all grids when there are alot but only subset of data', () => {
```

Comment:
I meant for this to mean testing lots of data sets...I'll update the name...I also realized I didn't actually test what I meant to test. Basically when a certain number of grids are loaded, it will load them all, but only load the data for what is showing on the screen.